whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java

```diff
@@ -0,0 +1,35 @@
+package net.ripe.db.whois.common.dao.jdbc.index;
+
+import net.ripe.db.whois.common.dao.RpslObjectInfo;
+import net.ripe.db.whois.common.rpsl.AttributeParser;
+import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import net.ripe.db.whois.common.rpsl.attrs.AttributeParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+class IndexWithEmailAddressValueAndType extends IndexWithValueAndType {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IndexWithEmailAddressValueAndType.class);
+
+    private static final AttributeParser.EmailParser EMAIL_PARSER = new AttributeParser.EmailParser();
+
+    public IndexWithEmailAddressValueAndType(final AttributeType attributeType, final String lookupTableName, final String lookupColumnName) {
+        super(attributeType, lookupTableName, lookupColumnName);
+    }
+
+    @Override
+    public int addToIndex(JdbcTemplate jdbcTemplate, RpslObjectInfo objectInfo, RpslObject object, String value) {
+        return super.addToIndex(jdbcTemplate, objectInfo, object, normaliseEmailAddress(value));
+    }
+
+    private String normaliseEmailAddress(final String emailAddress) {
+        try {
+            return EMAIL_PARSER.parse(emailAddress).getAddress();
```

Comment:
this function is  duplicated with  indexWithEmailAddressValue?

Reply:
There's no easy way to share this method as the classes extend different classes (the hierarchy is different)

Reply:
We have to extend `IndexWithValue` and `IndexWithValueAndType` with separate classes

Reply:
Updated code changes